### PR TITLE
fix: add publish config

### PR DESCRIPTION
**Pull request by jenniferarnesen**
Configs copied verbatim from @dhis2/analytics.
Added react, react-dom as peer dependencies
Other config changes in package.json for a proper published lib.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,9 +3,16 @@
...
     "license": "BSD-3-Clause",
-    "private": true,
     "main": "build/cjs/lib.js",
     "module": "build/es/lib.js",
+    "private": false,
+    "publishConfig": {
+        "access": "public"
+    },
+    "sideEffects": [
+        "./src/index.js"
```
Comment (amcgee) [inline package.json:13]: Is this necessary?

File: package.json [hunk 2]
```diff
@@ -29,6 +36,8 @@
     },
     "peerDependencies": {
         "@dhis2/app-runtime": "^2.0.4",
-        "d2": "^31.8.1"
+        "d2": "^31.8.1",
+        "react": "^16.13.1",
+        "react-dom": "^16.13.1"
```
Comment (amcgee) [inline package.json:41]: These should be `^16.8`, we don't rely on any later features of React

File: package.json [hunk 3]
```diff
@@ -29,6 +36,8 @@
     },
     "peerDependencies": {
         "@dhis2/app-runtime": "^2.0.4",
-        "d2": "^31.8.1"
+        "d2": "^31.8.1",
```
Comment (amcgee) [inline package.json:39]: These could probably be a bit more liberal - `d2@>30` maybe?  And perhaps `^2` for `app-runtime`?  I don't know if there's a minimum version we should enforce for `d2`...

File: .github/workflows/node-publish.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            - '[0-9]+.x.x'
+            - '[0-9]+.[0-9]+.x'
+
+env:
+    GIT_AUTHOR_NAME: '@dhis2-bot'
+    GIT_AUTHOR_EMAIL: 'apps@dhis2.org'
+    GIT_COMMITTER_NAME: '@dhis2-bot'
+    GIT_COMMITTER_EMAIL: 'apps@dhis2.org'
+    NPM_TOKEN: ${{secrets.DHIS2_BOT_NPM_TOKEN}}
+    GH_TOKEN: ${{secrets.DHIS2_BOT_GITHUB_TOKEN}}
```
Comment (amcgee) [inline .github/workflows/node-publish.yml:23]: I'll grant this repo access to the org secrets so that these will work

File: package.json (hunk 2 above)
Comment (jenniferarnesen) [reply package.json:41]: ok, I'll update.

File: package.json (hunk 1 above)
Comment (jenniferarnesen) [reply package.json:13]: Do you mean the sideEffects? or the `access: "public"`? (or both).

File: package.json (hunk 3 above)
Comment (jenniferarnesen) [reply package.json:39]: Tests failed with d2 ^30. Passed with d2 ^31. So I put it at 31.

File: package.json (hunk 1 above)
Comment (amcgee) [reply package.json:13]: sideEffects - are there actually non-tree-shakable sideEffects in that file...?

File: package.json (hunk 3 above)
Comment (amcgee) [reply package.json]: OK, could be a bit more lenient and leave it at `^31`, but not critical

Comment (amcgee) [reply package.json]: And actually it should be `>31` not `^31` because otherwise it won't match versions `>=32`

Comment (amcgee) [reply package.json]: (though that doesn't exist yet and might never exist... so not a huge deal)

File: package.json (hunk 1 above)
Comment (jenniferarnesen) [reply package.json:13]: @amcgee I thought that initializing the config is a side effect?

File: package.json (hunk 3 above)
Comment (jenniferarnesen) [reply package.json]: Updated to >30

File: package.json (hunk 1 above)
Comment (jenniferarnesen) [reply package.json]: Right.

File: package.json [hunk 5]
```diff
@@ -19,16 +24,18 @@
         "@testing-library/jest-dom": "^5.11.2",
         "@testing-library/react": "^10.4.8",
         "@testing-library/react-hooks": "^3.4.1",
+        "d2": ">30",
```
Comment (amcgee) [inline package.json:27]: I would recommend maybe setting the dev dependency to a more explicit version?

Comment (amcgee) [reply package.json:27]: Not critical

**Pull request merged**